sparkle/src/components/NewDropdown.tsx

```diff
@@ -170,38 +170,70 @@ const NewDropdownMenuContent = React.forwardRef<
 ));
 NewDropdownMenuContent.displayName = DropdownMenuPrimitive.Content.displayName;
 
+type NewDropdownMenuItemProps = MutuallyExclusiveProps<
+  React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Item> & {
+    inset?: boolean;
+    variant?: ItemVariantType;
+  } & Omit<LinkWrapperProps, "children" | "className">,
+  LabelAndIconProps & { description?: string }
+>;
+
 const NewDropdownMenuItem = React.forwardRef<
   React.ElementRef<typeof DropdownMenuPrimitive.Item>,
-  MutuallyExclusiveProps<
-    React.ComponentPropsWithoutRef<typeof DropdownMenuPrimitive.Item> & {
-      inset?: boolean;
-      variant?: ItemVariantType;
-    },
-    LabelAndIconProps & { description?: string }
-  >
+  NewDropdownMenuItemProps
 >(
   (
-    { children, variant, description, className, inset, icon, label, ...props },
+    {
+      children,
+      variant,
+      description,
+      className,
+      inset,
+      icon,
+      label,
+      href,
+      target,
+      rel,
+      asChild,
+      replace,
+      shallow,
+      ...props
+    },
     ref
-  ) => (
-    <DropdownMenuPrimitive.Item
-      ref={ref}
-      className={cn(
-        menuStyleClasses.item({ variant }),
-        inset ? menuStyleClasses.inset : "",
-        className
-      )}
-      {...props}
-    >
-      <ItemWithLabelIconAndDescription
-        label={label}
-        icon={icon}
-        description={description}
+  ) => {
+    const content = (
+      <DropdownMenuPrimitive.Item
+        ref={ref}
+        className={cn(
+          menuStyleClasses.item({ variant }),
+          inset ? menuStyleClasses.inset : "",
+          className
+        )}
+        {...props}
+        asChild={!!href && asChild}
```

Comment:
Can we reflect this limitation in the props with a proper exclusion type?

Reply:
Well we actually need a `asChild={!!href || asChild}` instead